LanguageFeatures/Private-named-parameters/static_semantics_A01_t05.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Given a named initializing formal or field parameter (for a
+/// primary constructor) with private name `p` in constructor `C`:
+/// - If `p` has no corresponding public name `n`, then compile-time error.
+///
+/// @description Check that it is a compile-time error if a named formal
+/// parameter of a declaring constructor has a private name with no
+/// corresponding public name.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=private-named-parameters,declaring-constructors
+
+class C1({var String __}) {
+//                   ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C2({required final String _}) {
+//                              ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C3 {
+  this({required final String _});
+//                            ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C4 {
+  this({var String __ = ""});
+//                 ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+extension type ET1 {
+  this({final String __ = ""});
+//                   ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+extension type ET2 {
+  this({final String _ = ""});
+//                   ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum E1({required final String __}) {
+//                             ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  e0(_: "");
```

Comment:
Good catch, too!

It makes sense to check that we don't accept funny invocations based on a wrong "corresponding public name" which isn't public.